src/main/java/com/google/sps/servlets/EventServlet.java

```diff
@@ -76,56 +90,37 @@ public void doGet(HttpServletRequest request, HttpServletResponse response) thro
+      System.out.println(gson.toJson(event));
     }
   }
 
   @Override
   public void doPost(HttpServletRequest request, HttpServletResponse response) throws IOException {
-    String jsonBody =
-        new BufferedReader(new InputStreamReader(request.getInputStream()))
-            .lines()
-            .collect(Collectors.joining("\n"));
-
-    // Error handling: return error if jsonBody is empty
-    if (jsonBody == null || jsonBody.trim().length() == 0) {
-      response.sendError(HttpServletResponse.SC_BAD_REQUEST, "empty JSON body");
-      return;
-    }
-
-    // Parse the jsonBody String to a Gson object
-    Gson gson = new Gson();
-    Type type = new TypeToken<Map<String, String>>() {}.getType();
-    Map<String, String> eventJsonAsMap = gson.fromJson(jsonBody, type);
-
-    // Check if all the fields are present
-    try {
-      checkAllEventFieldsPresent(eventJsonAsMap);
-    } catch (ServletException exception) {
-      response.sendError(HttpServletResponse.SC_BAD_REQUEST, exception.getMessage());
-      return;
-    }
-
     // Get the value entered in the form
-    String event_lat = Jsoup.clean(eventJsonAsMap.get("lat"), Safelist.basic());
-    Double event_Lat_D = Double.parseDouble(event_lat);
-    String event_lng = Jsoup.clean(eventJsonAsMap.get("lng"), Safelist.basic());
-    Double event_Lng_D = Double.parseDouble(event_lng);
-    String event_name = Jsoup.clean(eventJsonAsMap.get("name"), Safelist.basic());
-    String event_desc = Jsoup.clean(eventJsonAsMap.get("description"), Safelist.basic());
-    String event_loc = Jsoup.clean(eventJsonAsMap.get("loc"), Safelist.basic());
+    Double event_Lat_D = Double.parseDouble(request.getParameter("lat"));
+    Double event_Lng_D = Double.parseDouble(request.getParameter("lng"));
+    String event_name = Jsoup.clean(request.getParameter("name"), Safelist.basic());
+    String event_desc = Jsoup.clean(request.getParameter("description"), Safelist.basic());
+    String event_loc = Jsoup.clean(request.getParameter("location_name"), Safelist.basic());
     TimestampValue event_time =
-        TimestampValue.of(Timestamp.parseTimestamp((eventJsonAsMap.get("date"))));
-    String event_type = eventJsonAsMap.get("type");
-    String event_sub = eventJsonAsMap.get("subject");
-    Long event_school_L = Long.parseLong(eventJsonAsMap.get("school_id"));
+        TimestampValue.of(Timestamp.parseTimestamp(request.getParameter("utc-date")));
+    String event_type = request.getParameter("event_type");
+    String event_sub = request.getParameter("subject");
+    Long event_school_L = Long.parseLong(request.getParameter("school_id"));
 
     // Store to the Datastore
     Datastore datastore = DatastoreOptions.getDefaultInstance().getService();
     KeyFactory keyFactory = datastore.newKeyFactory().setKind("Event");
     LatLng position = LatLng.of(event_Lat_D, event_Lng_D);
-    FullEntity eventEntity =
+    FullEntity<IncompleteKey> eventEntity =
```

Comment:
out of curiosity: what's the benefit of adding `IncompleteKey`?

Reply:
This is to get rid of a warning that says FullEntity is a raw type. IncompleteKey specifies that the entity is created without an id given and datastore will generate it automatically.